src/utils.test.js

```diff
@@ -573,6 +578,175 @@ describe('isFormEditing', () => {
   });
 });
 
+describe('getNextSelectedRowsState', () => {
+  it('should add a row to selectedRows if not selected', () => {
+    const selectedRows = {};
+    const row = { id: 1, name: 'John Doe' };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(result).toEqual({ [row.id]: row });
+  });
+
+  it('should remove a row from selectedRows if already selected', () => {
+    const row = { id: 1, name: 'John Doe' };
+    const selectedRows = { [row.id]: row };
+
+    const result = getNextSelectedRowsState(selectedRows, row);
+
+    expect(result).toEqual({});
+  });
+
+  it('should not mutate the original selectedRows object', () => {
+    const row = { id: 1, name: 'John Doe' };
```

Comment:
Could you create constant in this file for `row`?
I see that you use the same object at least three times.